### PR TITLE
Add present_with_defaults to sync plans

### PR DESCRIPTION
**Pull request by bagasse**
Add present_with_defaults to sync plan to be able to avoid 'change loops' if we want to manage `sync plan <-> products` association in products and not sync plans.

### CODE REVIEW

File: plugins/modules/katello_sync_plan.py [hunk 1]
```diff
@@ -122,6 +124,7 @@ def main():
             enabled=dict(type='bool', required=True),
             sync_date=dict(required=True),
             cron_expression=dict(),
+            state=dict(default='present', choices=['present_with_defaults', 'present', 'absent']),
```
Comment (evgeni) [inline plugins/modules/katello_sync_plan.py]: I like that the only thing needed to support that is the addition to the `state` dict.

File: tests/test_playbooks/sync_plan.yml [hunk 2]
```diff
@@ -46,6 +46,15 @@
...
       sync_plan_cron_expression: "* * * * */5"
       expected_change: true
+  - include: tasks/product.yml
+    vars:
+      product_sync_plan: Test Sync Plan
+      product_state: present
+      expected_change: true
+  - include: tasks/sync_plan.yml
+    vars:
+      sync_plan_state: present_with_defaults
```
Comment (evgeni) [inline tests/test_playbooks/sync_plan.yml]: shouldn't this test pass some parameter that won't be updated? like the product list or something?

File: plugins/modules/katello_sync_plan.py (hunk 1 above)
Comment (bagasse) [reply plugins/modules/katello_sync_plan.py]: Yes ! I like this too !

File: tests/test_playbooks/sync_plan.yml (hunk 2 above)
Comment (bagasse) [reply tests/test_playbooks/sync_plan.yml]: Yes, you are right, and it permit me to find a bug. Changed.

**Pull request merged**